@here/harp-mapview/lib/geometry/LodMesh.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MathUtils } from "@here/harp-geoutils";
+import * as THREE from "three";
+
+/**
+ * A mesh that can store multiple versions of a geometry with differents level of detail.
+ * The rendered level of detail can be adapted during runtime.
+ * @internal
+ * @hidden
+ */
+export class LodMesh extends THREE.Mesh {
+    private m_geometries: Array<THREE.Geometry | THREE.BufferGeometry> | undefined;
+
+    /**
+     * Creates a [[LodMesh]] with given geometries and materials
+     * @param geometries A list of geometries with different levels of detail
+     * @param material Material for the mesh
+     */
+    constructor(
+        geometries?: Array<THREE.Geometry | THREE.BufferGeometry>,
+        material?: THREE.Material | THREE.Material[] | undefined
+    ) {
+        super(undefined, material);
+
+        this.geometries = geometries;
+    }
+
+    /**
+     * Update geometries of mesh
+     */
+    set geometries(geometries: Array<THREE.Geometry | THREE.BufferGeometry> | undefined) {
+        this.m_geometries = geometries;
+        this.geometry =
```

Comment:
What happens to the previously used geometry? I mean will the GL resources be freed?

Reply:
Added `dispose` if geometry is changed.